### DISCUSSION TITLE
Angular 12 Universal Application fails to add angularfire at step 2

### DISCUSSION THREAD

**Discussion by kveratis** [angular/angularfire#3007] [Q&A]
I'm building an Angular 12 Universal app and the process fails right at step 2 with ng add @angular/fire.  First it doesn't detect that the application is a universal app then right after installing the packages it exits with the code "ENOENT: no such file or directory, open '\C\Temp\dummy-universal\firebase.json'" This is on Windows 10.  I tried doing the process on my Ubuntu Linux build server and it also failed at the same point with the following error

✖ Package install failed.
/home/kveratis/dev/benevently-website/node_modules/rxjs/internal/util/hostReportError.js:4
setTimeout(function () { throw err; }, 0);
                             ^

TypeError: Cannot read property 'message' of undefined
    at Object.error (/home/kveratis/dev/benevently-website/node_modules/@angular/cli/models/schematic-command.js:432:47)
    at SafeSubscriber.__tryOrUnsub (/home/kveratis/dev/benevently-website/node_modules/rxjs/internal/Subscriber.js:205:16)
    at SafeSubscriber.error (/home/kveratis/dev/benevently-website/node_modules/rxjs/internal/Subscriber.js:156:26)
    at Subscriber._error (/home/kveratis/dev/benevently-website/node_modules/rxjs/internal/Subscriber.js:92:26)
    at Subscriber.error (/home/kveratis/dev/benevently-website/node_modules/rxjs/internal/Subscriber.js:72:18)
    at TapSubscriber._error (/home/kveratis/dev/benevently-website/node_modules/rxjs/internal/operators/tap.js:75:26)
    at TapSubscriber.Subscriber.error (/home/kveratis/dev/benevently-website/node_modules/rxjs/internal/Subscriber.js:72:18)
    at MergeMapSubscriber.SimpleOuterSubscriber.notifyError (/home/kveratis/dev/benevently-website/node_modules/rxjs/internal/innerSubscribe.js:72:26)
    at SimpleInnerSubscriber._error (/home/kveratis/dev/benevently-website/node_modules/rxjs/internal/innerSubscribe.js:30:21)
    at SimpleInnerSubscriber.Subscriber.error (/home/kveratis/dev/benevently-website/node_modules/rxjs/internal/Subscriber.js:72:18)

**Comment by jamesdaniels** [comment]
Hmmm...I'm not able to reproduce on a fresh angular app. Can you email me your package.json and angular.json? jamesdaniels@google.com

**Reply by kveratis** [reply]
I noticed that this only happens for version 7.1.0 if I specify going back to 7.0.4 everything works as expected.

package.json

{
  "name": "benevently-website",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "watch": "ng build --watch --configuration development",
    "test": "ng test",
    "dev:ssr": "ng run benevently-website:serve-ssr",
    "serve:ssr": "node dist/benevently-website/server/main.js",
    "build:ssr": "ng build && ng run benevently-website:server",
    "prerender": "ng run benevently-website:prerender"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "~12.2.0",
    "@angular/common": "~12.2.0",
    "@angular/compiler": "~12.2.0",
    "@angular/core": "~12.2.0",
    "@angular/fire": "^7.1.0",
    "@angular/forms": "~12.2.0",
    "@angular/platform-browser": "~12.2.0",
    "@angular/platform-browser-dynamic": "~12.2.0",
    "@angular/platform-server": "~12.2.0",
    "@angular/router": "~12.2.0",
    "@nguniversal/express-engine": "^12.1.1",
    "express": "^4.15.2",
    "rxjs": "~6.6.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.11.4",
    "firebase": "^9.1.0",
    "rxfire": "^6.0.0"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~12.2.9",
    "@angular/cli": "~12.2.9",
    "@angular/compiler-cli": "~12.2.0",
    "@nguniversal/builders": "^12.1.1",
    "@types/express": "^4.17.0",
    "@types/jasmine": "~3.8.0",
    "@types/node": "^12.11.1",
    "jasmine-core": "~3.8.0",
    "karma": "~6.3.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage": "~2.0.3",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "~1.7.0",
    "typescript": "~4.3.5"
  }
}

angular.json

{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "cli": {
    "analytics": "85d541e5-0de8-45f1-af50-254b4ae179ce"
  },
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "website": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:application": {
          "strict": true
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/website/browser",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.app.json",
            "assets": ["src/favicon.svg", "src/assets"],
            "styles": ["src/styles.css"],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "1mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "browserTarget": "website:build:production"
            },
            "development": {
              "browserTarget": "website:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "website:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "main": "src/test.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.spec.json",
            "karmaConfig": "karma.conf.js",
            "assets": ["src/favicon.svg", "src/assets"],
            "styles": ["src/styles.css"],
            "scripts": []
          }
        },
        "server": {
          "builder": "@angular-devkit/build-angular:server",
          "options": {
            "outputPath": "dist/website/server",
            "main": "server.ts",
            "tsConfig": "tsconfig.server.json",
            "externalDependencies": [
              "@nestjs/microservices",
              "@nestjs/microservices/microservices-module",
              "@nestjs/websockets",
              "@nestjs/websockets/socket-module",
              "cache-manager"
            ]
          },
          "configurations": {
            "production": {
              "outputHashing": "media",
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "optimization": false
            },
            "development": {
              "optimization": false,
              "sourceMap": true,
              "extractLicenses": false
            }
          },
          "defaultConfiguration": "production"
        },
        "serve-ssr": {
          "builder": "@nguniversal/builders:ssr-dev-server",
          "configurations": {
            "development": {
              "browserTarget": "website:build:development",
              "serverTarget": "website:server:development"
            },
            "production": {
              "browserTarget": "website:build:production",
              "serverTarget": "website:server:production"
            }
          },
          "defaultConfiguration": "development"
        },
        "prerender": {
          "builder": "@nguniversal/builders:prerender",
          "options": {
            "routes": ["/"]
          },
          "configurations": {
            "production": {
              "browserTarget": "website:build:production",
              "serverTarget": "website:server:production"
            },
            "development": {
              "browserTarget": "website:build:development",
              "serverTarget": "website:server:development"
            }
          },
          "defaultConfiguration": "production"
        }
      }
    }
  },
  "defaultProject": "website"
}

**Reply by miguelcamposdev** [reply]
Any solution for this issue? I've the same problem in my MacBook Pro with M1 chip and I've installed the @angular/fire v7.2.0

**Reply by onedapperterm** [reply]
Any solution? I have the same problem! i`ve tried everything! :/

**Reply by harshal0412** [reply]
facing the same issue in Macbook pro with intel i5

**Reply by onedapperterm** [reply]
Hey guys, I've figured out. The thing is that the installation of this library when we use Ionic is slightly different. At the end of the npm angular fire homepage is this link: 

https://github.com/angular/angularfire/blob/HEAD/docs/ionic/cli.md

That's the installation that worked for me using just this command:

```
npm install @angular/fire firebase --save
```

The deal here is, that with this command you are installing Angular fire 2 in a previous version, so you have to import your firebase module from the fire/compat folder instead of the actual fire folder. (At least for me worked like that)

**Reply by harshal0412** [reply]
> Hey guys, I've figured out. The thing is that the installation of this library when we use Ionic is slightly different. At the end of the npm angular fire homepage is this link:
> 
> https://github.com/angular/angularfire/blob/HEAD/docs/ionic/cli.md
> 
> That's the installation that worked for me using just this command:
> 
> ```
> npm install @angular/fire firebase --save
> ```
> 
> The deal here is, that with this command you are installing Angular fire 2 in a previous version, so you have to import your firebase module from the fire/compat folder instead of the actual fire folder. (At least for me worked like that)

thanks, it worked
